stdlib/3/ast.pyi

```diff
@@ -1,16 +1,16 @@
 # Python 3.5 ast
 
 import typing
-from typing import Any, Union, Iterator
+from typing import Any, Union, Iterator, Optional
 
 from _ast import *
 
 class NodeVisitor():
     def visit(self, node: AST) -> Any: ...
-    def generic_visit(self, node: AST) -> None: ...
+    def generic_visit(self, node: AST) -> Any: ...
 
 class NodeTransformer(NodeVisitor):
-    def generic_visit(self, node: AST) -> None: ...
+    def generic_visit(self, node: AST) -> Optional[AST]: ...
```

Comment:
Something I missed while initially looking at this: Why return `Optional`? It seems to be `generic_visit()` always returns the node passed in. The note in https://docs.python.org/3.7/library/ast.html about returning `None` refers to the `visit()` method, I think?

Reply:
`Optional` return is okay, as well as `List[AST]`.

See https://github.com/python/cpython/blob/c8bb467f4006fbf5d24d2491248bcbabee5d827e/Lib/ast.py#L310 .

The correct return type is `Union[AST, None, List[AST]]`.

Reply:
```python
value = self.visit(value) 
if value is None:
    continue
elif not isinstance(value, AST):
     new_values.extend(value)
     continue
```
Note that `self.visit` is invoked and the return is `value`, so `if value is None` and `new_values.extend(value)` could indicate the type exactly..

Reply:
Good point, I missed that.